core/editor/src/document/document_message_handler.rs

```diff
@@ -382,10 +387,26 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 					}
 				}
 			}
-			PasteLayers => {
-				for layer in self.copy_buffer.iter() {
-					//TODO: Should be the path to the current folder instead of root
-					responses.push_back(DocumentOperation::PasteLayer { layer: layer.clone(), path: vec![] }.into())
+			PasteLayers { path, insert_index } => {
+				let paste = |layer: &Layer, responses: &mut VecDeque<_>| {
+					log::trace!("pasting into folder {:?} as index: {}", path, insert_index);
+					responses.push_back(
+						DocumentOperation::PasteLayer {
+							layer: layer.clone(),
+							path: path.clone(),
+							insert_index,
+						}
+						.into(),
+					)
+				};
+				if insert_index == -1 {
+					for layer in self.copy_buffer.iter() {
+						paste(layer, responses)
+					}
+				} else {
+					for layer in self.copy_buffer.iter().rev() {
+						paste(layer, responses)
+					}
```

Comment:
This could be more readable and remove excess indentation scopes:
```rust
let layers = self.copy_buffer.iter();
let layers = if insert_index == -1 { layers } else { layers.rev() };
for layer in layers {
	paste(layer, responses)
}
```

Reply:
This does not work since layers.iter() has a different type than layers.iter().rev()